python_modules/libraries/dagstermill/dagstermill/asset_factory.py

```diff
@@ -122,6 +135,7 @@ def define_dagstermill_asset(
     group_name: Optional[str] = None,
     io_manager_key: Optional[str] = None,
     retry_policy: Optional[RetryPolicy] = None,
+    save_notebook_on_failure: bool = False,
```

Comment:
should this be `Optional[bool]` instead?

Reply:
what are you thinking would be the behavior if it's None?

Reply:
`opt_bool_param` should convert None to false by default

Reply:
so the idea is user-facing wise, they can omit this arg but once we validate user inputs, all the args to `_dm_compute` should be not null.

Reply:
hmm i was mistaken `bool = False` and `Optional[bool] = False` both would work.